src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,105 @@
-import React from 'react';
+import React, {
+  useContext,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { StateContext } from './AppContext';
+import { addComment, getComments } from '../utils/loadutil';
+import { ACTIONS } from '../utils/enums';
 
 export const NewCommentForm: React.FC = () => {
+  const { state, dispatch } = useContext(StateContext);
+  const initialComment = {
+    name: '',
+    body: '',
+    email: '',
+    postId: state.selectedPost.id,
+  };
+
+  const [newComment, setNewComment] = useState(initialComment);
+
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isCommentTextError, setIsCommentTextError] = useState(false);
+
+  function hanldeInputName(e: React.ChangeEvent<HTMLInputElement>) {
+    e.preventDefault();
+    setNewComment({
+      ...newComment,
+      name: e.currentTarget.value,
+    });
+    setIsNameError(false);
+  }
+
+  function hanldeInputEmail(e: React.ChangeEvent<HTMLInputElement>) {
+    e.preventDefault();
+    setNewComment({
+      ...newComment,
+      email: e.currentTarget.value,
+    });
+    setIsEmailError(false);
+  }
+
+  function hanldeInputCommentText(e: React.ChangeEvent<HTMLTextAreaElement>) {
+    e.preventDefault();
+    setNewComment({
+      ...newComment,
+      body: e.currentTarget.value,
+    });
+    setIsCommentTextError(false);
+  }
+
+  function handleClear() {
+    setNewComment(initialComment);
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    if (newComment.body.length === 0) {
+      setIsCommentTextError(true);
+    }
+
+    if (newComment.name.length === 0) {
+      setIsNameError(true);
+    }
+
+    if (newComment.email.length === 0) {
+      setIsEmailError(true);
+    }
+    /* eslint-disable */
+    // disabling next eslint rules : unnecessary return statement no-useless-return Unnecessary 'else' after 'return'. I need these return & else.
```

Comment:
remove this disable and remove else belove

Reply:
+